test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,318 @@
+                    restartCount++;
+
+                    // Setup Message Task
+                    Task sendMessageTask;
+                    if (Settings.Current.MessageEnable)
+                    {
+                        Func<Task> sendMessage =
+                            async () =>
+                            {
+                                (DateTime msgCompletedTime, HttpStatusCode msgStatusCode) = await SendMessageAsync(
+                                    msgModuleClient,
+                                    Settings.Current.TrackingId,
+                                    batchId,
+                                    Settings.Current.MessageOutputEndpoint,
+                                    eachTestExpirationTime,
+                                    cts.Token).ConfigureAwait(false);
+                                TestResultBase msgTestResult = CreateTestResult(
+                                    TestOperationResultType.Messages,
+                                    restartTime,
+                                    restartStatus,
+                                    msgCompletedTime,
+                                    msgStatusCode,
+                                    batchId,
+                                    restartCount,
+                                    Interlocked.Read(ref messageCount));
+                                var reportClient = new TestResultReportingClient { BaseUrl = Settings.Current.ReportingEndpointUrl.AbsoluteUri };
+                                await ModuleUtil.ReportTestResultAsync(
+                                    reportClient,
+                                    Logger,
+                                    msgTestResult,
+                                    cts.Token).ConfigureAwait(false);
+                            };
+                        sendMessageTask = sendMessage();
+                    }
+                    else
+                    {
+                        sendMessageTask = Task.CompletedTask;
+                    }
+
+                    // Setup Direct Method Task
+                    Task directMethodTask;
+                    if (Settings.Current.DirectMethodEnable)
+                    {
+                        Func<Task> directMethod =
+                            async () =>
+                            {
+                                (DateTime dmCompletedTime, HttpStatusCode dmStatusCode) = await SendDirectMethodAsync(
+                                    Settings.Current.DeviceId,
+                                    Settings.Current.DirectMethodTargetModuleId,
+                                    dmModuleClient,
+                                    Settings.Current.DirectMethodName,
+                                    testExpirationTime,
+                                    cts.Token).ConfigureAwait(false);
+                                TestResultBase dmTestResult = CreateTestResult(
```

Comment:
what do you think if move line128-142 insdie SendDirectMethodAsync?  then you simplify here.

Reply:
Then `SendDirectMethodAsync()` will no long doing only the `Send direct method` and plus the input argument to the function is going to be large. I don't like that.